website/src/components/DataTable/useCursorPagination.ts

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+
+export interface CursorPaginationState {
+  /**
+   * The user's `display_name` used for pagination.
+   */
+  cursor: string;
+
+  /**
+   * The pagination direction.
+   */
+  direction: "forward" | "back";
+}
+
+export const useCursorPagination = () => {
+  const [pagination, setPagination] = useState<CursorPaginationState>({ cursor: "", direction: "forward" });
+
+  const toPreviousPage = (data: undefined | { prev?: string; next?: string }) => {
+    setPagination({
+      cursor: data?.prev || "",
+      direction: "back",
+    });
+  };
+
+  const toNextPage = (data: undefined | { prev?: string; next?: string }) => {
+    setPagination({
+      cursor: data?.next || "",
+      direction: "forward",
+    });
+  };
+
+  const resetCursor = () => setPagination((old) => ({ ...old, cursor: "" }));
```

Comment:
`useCallback` for these 3 functions.

Reply:
These 3 function will need a wrapper function to pass data, so useCallback has no effect here.